src/option.h

```diff
@@ -5,82 +5,115 @@
-enum lineends_e
+//-----------------------------------------------------------------------------
+/// I/A/R/F values - these are bit fields
+enum class iarf_e
 {
-   LE_LF,      //! "\n"   typically used on Unix/Linux system
-   LE_CRLF,    //! "\r\n" typically used on Windows systems
-   LE_CR,      //! "\r"   carriage return without newline
-   LE_AUTO     //! keep last
+   IGNORE      = 0,              //! option ignores a given feature
+   ADD         = (1u << 0),      //! option adds a given feature
+   REMOVE      = (1u << 1),      //! option removes a given feature
+   FORCE       = (ADD | REMOVE), //! option forces the usage of a given feature
+   NOT_DEFINED = (1u << 2)       //! for debugging
 };
 
-//! Token position - these are bit fields
-enum tokenpos_e
+UNC_DECLARE_FLAGS(iarf_flags_t, iarf_e)
+UNC_DECLARE_OPERATORS_FOR_FLAGS(iarf_flags_t)
+
+//-----------------------------------------------------------------------------
+/// Line endings
+enum class lineend_e
 {
-   TP_IGNORE      = 0,                        //! don't change it
-   TP_BREAK       = 1,                        //! add a newline before or after the if not present
-   TP_FORCE       = 2,                        //! force a newline on one side and not the other
-   TP_LEAD        = 4,                        //! at the start of a line or leading if wrapped line
-   TP_LEAD_BREAK  = (TP_LEAD | TP_BREAK),
-   TP_LEAD_FORCE  = (TP_LEAD | TP_FORCE),
-   TP_TRAIL       = 8,                        //! at the end of a line or trailing if wrapped line
-   TP_TRAIL_BREAK = (TP_TRAIL | TP_BREAK),
-   TP_TRAIL_FORCE = (TP_TRAIL | TP_FORCE),
-   TP_JOIN        = 16,                       //! remove newlines on both sides
+   LF,   //! "\n"   typically used on Unix/Linux system
+   CRLF, //! "\r\n" typically used on Windows systems
+   CR,   //! "\r"   carriage return without newline
+   AUTO, //! keep last
 };
+constexpr auto lineend_styles = static_cast<size_t>(lineend_e::AUTO);
 
-//! True, False or Ignore
-enum TrueFalseIgnore_e
+//-----------------------------------------------------------------------------
+/// Token position - these are bit fields
+enum class tokenpos_e
 {
-   TFI_FALSE  = 0,                    //! false
-   TFI_TRUE   = 1,                    //! true
-   TFI_IGNORE = 2,                    //! ignore
+   IGNORE      = 0,  //! don't change it
+   BREAK       = 1,  //! add a newline before or after the if not present
+   FORCE       = 2,  //! force a newline on one side and not the other
+   LEAD        = 4,  //! at the start of a line or leading if wrapped line
+   TRAIL       = 8,  //! at the end of a line or trailing if wrapped line
+   JOIN        = 16, //! remove newlines on both sides
+   LEAD_BREAK  = (LEAD | BREAK),
+   LEAD_FORCE  = (LEAD | FORCE),
+   TRAIL_BREAK = (TRAIL | BREAK),
+   TRAIL_FORCE = (TRAIL | FORCE),
 };
 
+UNC_DECLARE_FLAGS(tokenpos_flags_t, tokenpos_e)
+UNC_DECLARE_OPERATORS_FOR_FLAGS(tokenpos_flags_t)
+
+// TODO: generate these
```

Comment:
This is ugly as all else (and some of the `AT_*` intentionally, in order to avoid unnecessary code churn for now, don't match the actual enum values, which have been tweaked), but #1852 will make both issues go away. (Partly, through much less use of `AT_*` — which also is `OT_*` in #1852 — and also via `make_option_enum.py`... but I wanted to leave the improved enum conversions in #1852, and didn't want to try to take just the alias generation bits out of `make_option_enum.py`.)